assignment.md

```diff
@@ -12,46 +12,86 @@ If the version control concepts or `git` commands are still confusing to you, co
 
-    > Your answer here
+    > When I try to push the changes to 'upstream' it denies access because I don't have permission to push to that remote. The benefits of having the 'upstream' remote when working collaboratively is that you can pull other people's work.
 
 
  4. Now clone the repository for the [class website](https://github.com/WCM-datascibasics/wcm-datascibasics.github.io), and in the class website repository:
     - a. Explore the version history by visualizing it as a graph (remember the dog meme?).
         ```bash
         # git command and output here
+        git log --all --decorate --oneline --graph
+
+        * 1446124 (HEAD -> master, origin/master, origin/HEAD) add week 6
+* 540a90d added week5
+* a3f66fd swapped 2 weeks
+* cd0d211 python notes repo link
+* df90802 updated syllabus with presidents day
+* e0029df place holder schedule
+* 7462142 update lecture link for week 3
+* 8fced37 assignment 2 updated
+* 2a2bfd5 version control chapter
+* 13e5722 minor improvements to instructions
+* dd4719c fixed assignment 1 link
+* 0392554 fixed syllabus
+* 48e689c updated syllabus with MLK day
+* e96eba0 added dave's slides link
+* fbb9e8d fixed typo
+* c152f3c added ch2 assignment
+* cd40b26 fixed typo in assignment 1
+* b3455f6 fixed missing link
+* 7cd4587 update links to reflect erlative links
+* fb12bef jekyll config additions for relative links
+* 7ed94e9 dash fix
+* d6f3ec1 fixed link error
+* 2c05357 update links
         ```
 
     - b. Who was the last person to modify `README.md`? (Hint: use `git log` with an argument)
         ```bash
         # git command and output here
+        git log README.md
+commit 00e8ae51a08a0f3754ab228f62b51651595c4426
+Author: Xihe Xie <axiezai@gmail.com>
+Date:   Mon Dec 28 16:01:51 2020 -0500
+
         ```
     
     - c. What was the commit message associated with the last modification to the `-Week 2` line of `index.md` file? (Hint: use `git blame` and `git show`, check out their help pages if you are not sure what to do)
 
+    The commit message was "added ch2 assignment."
+
  5. In the course website repository (or another repository you want to clone from GitHub), modify one of its existing files, do not `git add` or `git commit` yet:
      - a. What happens when you do `git stash`, be specific about what happens to your uncommitted changes after `stash`.?
-       > Answer here
+       > It saves uncommitted changes in the stash log and removes them from the current directory. 
      - b. What do you see when you run `git log --all --oneline` after stashing your modification?
-       > Answer here
+       > My stash is shown on top of the log.
```

Comment:
Top of the log output, but indicated that it's separate from your git's history. So it's not committed.